addons/hr/models/hr_employee.py

```diff
@@ -340,23 +340,32 @@ def _get_user_creation_notification_action(message, message_type, next_action):
                     'type': 'ir.actions.client',
                     'tag': 'display_notification',
                     'params': {
-                        'title': _("User Creation Notification"),
+                        'title': self.env._("User Creation Notification"),
                         'type': message_type,
                         'message': message,
                         'next': next_action
                     }
                 }
 
+        normalize_users_email = [
+            email_normalize(user.email)
+            for user in self.env['res.users'].search([])
+            if user.email
+        ]
```

Comment:
I don't think you have to search for our several thousands users before checking if an email exists in db :disguised_face:  Always think with performance in mind: always try to narrow your search to speedup things, avoid fetching useless content.

Idea here would be
```suggestion
    employee_emails = [
        normalized_email
        for normalized_email in tools.mail.email_normalize_all(employee.work_email)
        for employee in self
    ]
    conflicting_users = self.env['res.users']
    if employee_emails:
        conflicting_users = self.env['res.users'].search([
            '|', ('email_normalized', 'in', employee_emails),
            ('login', 'in', employee_emails)
        ])
```